source/guides/maintenance/container-image-modify.rst

```diff
@@ -0,0 +1,477 @@
+
+Examples
+********
+
+Example 1: Add a bundle
+=======================
+
+In this example, we add :command:`wget` to the **clearlinux/redis**
+Dockerfile.
+
+#. Open a Terminal on |CL| and enter the :file:`dockerfiles/redis` directory.
+
+#. Enter :command:`swupd search wget` to discover which |CL| bundle includes
+   the software.
+
+#. Open a an editor to modify the Dockerfile with
+   :command:`$EDITOR Dockerfile`.
+
+#. Append the :command:`wget` bundle to the :command:`--bundles=` parameter
+   of the :command:`swupd os-install` command.
+
+#. Run :command:`git diff`.
+
+   The output shows the edits made after adding :command:`wget` in the clearlinux/redis Dockerfile.
+
+   .. code-block:: diff
+
+      diff --git a/redis/Dockerfile b/redis/Dockerfile
+      index af977cb..b1effab 100644
+      --- a/redis/Dockerfile
+      +++ b/redis/Dockerfile
+      @@ -15,7 +15,7 @@ RUN source /os-release && \
+          mkdir /install_root \
+          && swupd os-install -V ${VERSION_ID} \
+          --path /install_root --statedir /swupd-state \
+       -    --bundles=redis-native,findutils,su-exec --no-boot-update
+       +    --bundles=redis-native,findutils,su-exec,wget --no-boot-update
+
+#. Build the Dockerfile and apply a unique tag name. In this this example,
+   we use :command:`wget_added` and add proxies.
+
+   .. code-block:: bash
+
+      docker build \
+      --no-cache \
+      --build-arg http_proxy=$http_proxy \
+      --build-arg https_proxy=$https_proxy \
+      -t clearlinux/redis:wget_added .
+
+#. Run the Dockerfile with the `wget --version` command to verify that
+   :command:`wget` has been added to the image.
+
+   .. code-block:: bash
+
+      docker run clearlinux/redis:wget_added wget --version
+
+#. The output shows:
+
+   .. code-block:: console
+
+      GNU Wget 1.20.3 built on linux-gnu.
+
+      -cares +digest -gpgme +https +ipv6 -iri +large-file -metalink +nls
+      -ntlm +opie -psl +ssl/openssl
+
+
+Example 2: Change |CL| version (single-stage Dockerfiles)
+=========================================================
+
+This example shows how to rebuild single-stage containers against a specific
+OS version, :file:`<CL_VERSION>`, by adding a new argument to the Docker build command line.
+
+#. Open a Terminal on |CL| and enter the
+   :file:`dockerfiles/machine-learning-ui` directory.
+
+#. Rebuild the :file:`clearlinux/machine-learning-ui`. Add an extra build
+   argument :command:`swupd_args="-m <CL_VERSION>"`; in this case, the build version is 31090.
+
+   .. code-block:: bash
+      :linenos:
+      :emphasize-lines: 6
+
+      docker build \
+      --no-cache \
+      --build-arg http_proxy=$http_proxy \
+      --build-arg https_proxy=$https_proxy \
+      -t clearlinux/machine-learning-ui:31090 \
+      --build-arg swupd_args="-m 31090" .
+
+#. Run the docker container image:
+
+   .. code-block:: bash
+
+      docker run clearlinux/machine-learning-ui:31090 swupd info
```

Comment:
@puneetse, this is the last remaining problem. I've run step 2 above twice now. On Step 3, I receive the [same error message that was logged as this issue last March](https://github.com/clearlinux/distribution/issues/524). If you can supply stdout for sample output, that'd be great.

Reply:
@puneetse, please see my comment above. This is the last piece of data I need before we can merge.

Reply:
The command above worked fine for me. It should look like the output of any other swupd info with 31090

```
docker run clearlinux/machine-learning-ui:31090 swupd info
Distribution:      Clear Linux OS
Installed version: 31090
Version URL:       https://cdn.download.clearlinux.org/update
Content URL:       https://cdn.download.clearlinux.org/update
```


Or, we can change it to this so that it matches the pattern in a previous section.:
```
docker run clearlinux/machine-learning-ui:31090 cat /usr/lib/os-release
NAME="Clear Linux OS"
VERSION=1
ID=clear-linux-os
ID_LIKE=clear-linux-os
VERSION_ID=31090
PRETTY_NAME="Clear Linux OS"
ANSI_COLOR="1;35"
HOME_URL="https://clearlinux.org"
SUPPORT_URL="https://clearlinux.org"
BUG_REPORT_URL="mailto:dev@lists.clearlinux.org"
PRIVACY_POLICY_URL=http://www.intel.com/privacy
```

Reply:
@puneetse. Please review latest commit.